week4_assignment/src/components/Myhobby.tsx

```diff
@@ -0,0 +1,94 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import "./Myhobby.css";
+import Emptypage from "./Emptypage";
+
+interface HobbyResponse {
+  result: {
+    hobby: string;
+  };
+}
+
+export default function MyHobby() {
+  const [hobby, setHobby] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [otherHobby, setOtherHobby] = useState<string | null>(null);
+  const [memberNo, setMemberNo] = useState<string>("");
```

Comment:
```suggestion
  const [hobby, setHobby] = useState("");
  const [error, setError] = useState("");
  const [otherHobby, setOtherHobby] = useState("");
  const [memberNo, setMemberNo] = useState("");
```

초기값을 지정해준다면 타입을 지정하지 않아도 괜찮습니다!
이 경우에 초기값을 null로 지정하는 바람에 타입이 `<string | null>`이 되어버렸는데요,
hobby, error, otherHobby 등이 string 타입을 갖는다면, 위와 같이 초기값을 ""로 지정해주고, 별도의 타입지정은 해주지 않아도 됩니다.

추가로, memberNo도 마찬가지로, 초기값을 이용해서 타입을 추론할 수 있는 경우에는 굳이 명시적으로 타입을 지정해주지 않는 것이 더 낫다고 합니다!